dashboard/app/controllers/registrations_controller.rb

```diff
@@ -389,7 +389,7 @@ def existing_account
   #
   def edit
     @permission_status = current_user.child_account_compliance_state
-    @personal_account_linking_enabled = Policies::ChildAccount.can_link_new_personal_account?(current_user) && !(current_user.country_code.nil? && current_user.us_state.nil?)
+    @personal_account_linking_enabled = Policies::ChildAccount.can_link_new_personal_account?(current_user) && (!current_user.student? || !(current_user.country_code.nil? && current_user.us_state.nil?))
```

Comment:
This might be easier to understand if it was:

```
@personal_account_linking_enabled = Policies::ChildAccount.can_link_new_personal_account?(current_user) && !(current_user.student? && (current_user.country_code.nil? || current_user.us_state.nil?))
```

But maybe it is just as messy.

Reply:
Would it be simpler if all this logic was inside `can_link_new_personal_account`?

Reply:
I could do that.